### ISSUE TITLE
Help: The system has no response when cancel train with multi-gpu

### ISSUE THREAD

**Issue by chendd14** [BVLC/caffe#3538]
ubuntu 14.04LTS
Titan X * 8
cuda 7.5

I successfully built the newest caffe， but failed to runtest (RongHang said it's may be a bug with multi-gpu). When I tried to train a net with multi-gpu, though one of the gpus' cost was about 100%, the training process seemed not going on(no more training infomation was print). When I tried to stop it with Ctrl+C, the system halt, even could not reboot the system with "sudo shutdown -r now".

Forgive my broken English, thanks for your help.

**Comment by ShaggO** [comment]
I've experienced a similar problem on a similar setup (ubuntu 14.04LTS, Titan X * 2, CUDA 7.5)
I ran out of memory on the GPUs while training and got the out-of-memory message, but the program (probably) kept on waiting for a signal from the GPU.
It meant that all GPU calls by other processes were stalled, I was unable to kill the processes accessing the GPUs (even with "sudo kill -9 <PID>"), and that a hardware reboot was necessary ("sudo shutdown -r now" wasn't enough).

**Comment by chendd14** [comment]
@ShaggO Did you solve the problem? I tried with CUDA7.0, the problem remained. It is a waste of computing resources when train with only a GPU.

**Comment by chendd14** [comment]
@ShaggO Maybe it's a problem with CUDA, a similar problem occurred with TensorFlow. I'm going to change my OS to CentOS.

**Comment by ShaggO** [comment]
@chendd14 unfortunately I'm not in a position where I'm able to do much debugging myself since my GPUs are located at a cluster which other people use as well, and hence it's quite a serious problem if the machines have to be reset every now and then. I'm however keen to hear if the problem persists on CentOS.

**Comment by lfz** [comment]
exactly the same hardware and software, and the same error

**Comment by chendd14** [comment]
@lfz if you have more than 4 gpu devices in your server, you can reduce them to no more than 4, it will help

**Comment by lfz** [comment]
@chendd14 you mean physically reduced? or reduce it in command --gpu 1,2,3,4?

**Comment by chendd14** [comment]
@lfz physically, yes

**Comment by lfz** [comment]
@chendd14 Thank you~ so you split your server into 2? have you figured out what is the reason of this bug?

**Comment by chendd14** [comment]
@lfz yes I buy a new server.  I'm not familiar with cuda, so I donot figure out the reason.
